src/finch/tensor.py

```diff
@@ -383,11 +383,12 @@ def to_storage(self, storage: Storage) -> "Tensor":
         return Tensor(self._from_other_tensor(self, storage=storage))
 
     @classmethod
-    def _from_other_tensor(cls, tensor: "Tensor", storage: Storage | None) -> JuliaObj:
+    def _from_other_tensor(cls, tensor: "Tensor", storage: Storage) -> JuliaObj:
         order = cls.preprocess_order(storage.order, tensor.ndim)
-        return jl.swizzle(
-            jl.Tensor(storage.levels_descr._obj, tensor._obj.body), *order
+        result = jl.copyto_b(
+            jl.swizzle(jl.Tensor(storage.levels_descr._obj), *order), tensor._obj
         )
+        return jl.dropfills(result)
```

Comment:
I'd remove the `jl.dropfills` call unless `tensor` is dense, which is really the only case where it matters much.

Reply:
Done!